src/platform_impl/linux/x11/window.rs

```diff
@@ -323,6 +324,7 @@ impl UnownedWindow {
         let mut window = UnownedWindow {
             xconn: Arc::clone(xconn),
             xwindow: xwindow as xproto::Window,
+            visual,
```

Comment:
IIRC `visual` can be `x11rb::COPY_FROM_PARENT`, which isn't a valid visual ID; it's just a special value passed to `create_window` and other functions to indicate that the real visual should be copied from the parent. You need to use `get_window_attributes` to get the real visual ID if you don't know it already.